src/gui/dialogs/migrate_version_selection.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+   Copyright (C) 2008 - 2018
+   Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include "gui/dialogs/migrate_version_selection.hpp"
+
+#include "addon/manager_ui.hpp"
+#include "filesystem.hpp"
+#include "game_version.hpp"
+#include "gettext.hpp"
+#include "gui/auxiliary/find_widget.hpp"
+#include "gui/dialogs/message.hpp"
+#include "gui/widgets/listbox.hpp"
+#include "gui/widgets/window.hpp"
+#include "preferences/credentials.hpp"
+#include "preferences/game.hpp"
+
+#include <boost/algorithm/string.hpp>
+
+static lg::log_domain log_version_migration{"gui/dialogs/migrate_version_selection"};
+#define ERR_LOG_VERSION_MIGRATION   LOG_STREAM(err,   log_version_migration)
+#define WRN_LOG_VERSION_MIGRATION   LOG_STREAM(warn,  log_version_migration)
+#define LOG_LOG_VERSION_MIGRATION   LOG_STREAM(info,  log_version_migration)
+#define DBG_LOG_VERSION_MIGRATION   LOG_STREAM(debug, log_version_migration)
+
+namespace gui2::dialogs
+{
+REGISTER_DIALOG(migrate_version_selection)
+
+void migrate_version_selection::execute()
+{
+    migrate_version_selection mig = migrate_version_selection();
+    if(mig.versions_.size() > 0)
+    {
+        mig.show();
+    }
+}
+
+migrate_version_selection::migrate_version_selection()
+{
+    version_info current_version = game_config::wesnoth_version;
+    std::string current_version_str = filesystem::get_version_path_suffix();
+
+    for(unsigned int i = 1; i < current_version.minor_version(); i++)
```

Comment:
As it's a new file, please run it through clang-format.

Reply:
If you do that though, do a quick manual check of the file to make sure it didn't do anything stupid to the formatting.

Reply:
I'm not sure why it's doing any of the things it's currently configured to do anyway, so I'm not sure what would be a case of it doing something we don't want it to do.

Reply:
I'm not sure either, but it can happen. There will always be cases where clang-format gets things wrong, either because it's a case we didn't think of and thus didn't explicitly configure correctly, or because it's an unusual case where we actually _want_ non-standard formatting.

Automated formatting is fine and all, but blinding committing what it did without even double-checking is not a good idea.

Reply:
Also, is the current .clang-format even something that we want applied without discussing what the format should be?  It exists, but clearly hasn't been updated in quite a while (it still lists the C++ language version as C++11), and I have no idea currently what changes it's set to make or if they're actually desirable.

Reply:
I suppose also that it doesn't really make a lot of sense to me to only apply clang-format to one file at a time.  Code formatting isn't really something that consistent across the rest of the codebase anyway, so only applying here and there occasionally feels like it'll just end up as an instance of https://xkcd.com/927/

Reply:
A new file is code that wasn't there before, so running the code formatter on it doesn't add extra changes to the history. Running the code formatter on existing files does generate extra changes.

The main thing I noticed in this file was the placement of opening braces being inconsistent with other Wesnoth files. Our format may be a bit odd, but the current codebase is consistent enough to notice.

Reply:
Updated, I suppose.  I'm not a fan of how it adjusts comments especially though, and the line length limit generally.